Mage.Sets/src/mage/cards/z/ZurEternalSchemer.java

```diff
@@ -0,0 +1,128 @@
+    }
+
+    public ZurEternalSchemer(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{W}{U}{B}");
+
+        addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(4);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Enchantment creatures you control have deathtouch, lifelink, and hexproof.
+        CompoundAbility compoundAbilities = new CompoundAbility(DeathtouchAbility.getInstance(), LifelinkAbility.getInstance(), HexproofAbility.getInstance());
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityControlledEffect(compoundAbilities, Duration.WhileOnBattlefield, filter)));
+
+        // {1}{W}: Target non-Aura enchantment you control becomes a creature in addition to its other types and has base power and toughness each equal to its mana value.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new ZurEternalSchemerEffect(), new ManaCostsImpl<>("{1}{W}"));
+        ability.addTarget(new TargetPermanent(filter2));
+        this.addAbility(ability);
+    }
+
+    private ZurEternalSchemer(final ZurEternalSchemer card) {
+        super (card);
+    }
+
+    @Override
+    public ZurEternalSchemer copy() {
+        return new ZurEternalSchemer(this);
+    }
+}
+
+class ZurEternalSchemerEffect extends ContinuousEffectImpl {
+
+    public ZurEternalSchemerEffect() {
+        super(Duration.EndOfGame, Outcome.BecomeCreature);
+        staticText = "Target non-Aura enchantment you control becomes a creature in addition to its other types " +
+                "and has base power and base toughness each equal to its mana value.";
+    }
+
+    public ZurEternalSchemerEffect(final ZurEternalSchemerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ZurEternalSchemerEffect copy() {
+        return new ZurEternalSchemerEffect(this);
+    }
+
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
+        Permanent enchantment = game.getPermanent(this.getTargetPointer().getFirst(game, source));
+        if (enchantment == null) {
+            return false;
+        }
+        switch (layer) {
+            case TypeChangingEffects_4:
+                if (sublayer == SubLayer.NA) {
+                    if (!enchantment.isCreature(game)) {
+                        enchantment.addCardType(game, CardType.CREATURE);
+                    }
+                }
+                break;
+
+            case PTChangingEffects_7:
+                if (sublayer == SubLayer.SetPT_7b) {
+                    int cmc = enchantment.getManaValue();
+                    enchantment.getPower().setValue(cmc);
+                    enchantment.getToughness().setValue(cmc);
+                }
+        }
+        return true;
```

Comment:
I think you should return true only if either the type change or PT change has occurred

Reply:
As in changing line 115 to `return false` and having `return true` within the if blocks of lines 102 and 109? In testing this yields different results.
As the code is now, an enchantment creature ([[Dockside Chef]] in this case, which is a 1 mana 1/2 creature) being affected by [[Weaver of Harmony]] and [[Opalescence]] has it as a 2/2, as its base stats are set to 1/1 by Opalescence, and its stats are then buffed by +1/+1 by the Weaver of Harmony:
![before](https://user-images.githubusercontent.com/26198472/186491496-0be1a0c6-c767-40e3-858e-740da1807a83.PNG)

With the changes I outlined above, however, the Dockside Chef is a 2/3, only being affected by the Weaver of Harmony and unaffected by Opalescence. Sacrificing the Weaver of Harmony also revealed that its base stats were still 1/2, and were not thus being affected by Opalescence even though it fits the "non-Aura enchantment" description.
![after](https://user-images.githubusercontent.com/26198472/186492045-814dad5d-3187-491d-95ba-239d783a4829.PNG)
Am I misunderstanding / misimplementing the requested changes or?

Reply:
> As in changing line 115 to `return false` and having `return true` within the if blocks of lines 102 and 109? In testing this yields different results.
Not exactly what I mean.

Define something like `boolean applied = false`

And then on `103` and `112` do a `applied = true`, and at the end return `applied`. This way both effects are checked and potentially applied, and you return `false` if neither were applied and `true` if either of them were.

Reply:
This doesn't seem to work for Opalescence and Starfield of Nyx, as the effect seems to be permanently applied to affected permanents even when Opalescence leaves the battlefield. Code below is the new Opalescence `apply` code I tested (whch doesn't work):

```
public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
    boolean applied = false;
    for (Permanent permanent : game.getBattlefield().getActivePermanents(filter, 
            source.getControllerId(), source, game)) {
        applied = false;
        switch (layer) {
            case TypeChangingEffects_4:
                if (sublayer == SubLayer.NA) {
                    if (!permanent.isCreature(game)) {
                        permanent.addCardType(game, CardType.CREATURE);
                        applied = true;
                    }
                }
                break;

            case PTChangingEffects_7:
                if (sublayer == SubLayer.SetPT_7b) {
                    int manaCost = permanent.getManaValue();
                    permanent.getPower().modifyBaseValue(manaCost);
                    permanent.getToughness().modifyBaseValue(manaCost);
                    applied = true;
                }
        }

    }
    return applied;
}
```

Reply:
Looking at MageInt, I would except the effect to continue indefinitely because of how resetToBaseValue() is currently implemented.

I think this is the issue @theelk801 was talking about for Bess: https://github.com/magefree/mage/pull/9328#discussion_r946749106

Reply:
Ah I see. Maybe this PR should be put on hold then until base PT values have been refactored. I notice you have a PR open (#9409) so maybe that will be relevant to resolving this?

Reply:
> Ah I see. Maybe this PR should be put on hold then until base PT values have been refactored. I notice you have a PR open (#9409) so maybe that will be relevant to resolving this?

Yeah, I think we should wait until that's done before adding more cards that care about base P/T.

Yes, I've started working on. So I guess keep an eye there for when that's done

Reply:
@PurpleCrowbar, the base P/T PR is merged, you're good to continue with this

Reply:
The interaction that you're showing in the screenshot above, could you please add that as a test to make sure that the interaction continues to work properly going forward.

Reply:
> The interaction that you're showing in the screenshot above, could you please add that as a test to make sure that the interaction continues to work properly going forward.

Added a test for Opalescence as that was the problem card, the primary issue being that its effect of turning enchantments into creatures wouldn't end when it left the battlefield. First time writing a test but passes fine and seems to be correct